force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,42 @@
+trigger AccountTrigger on Account (before insert, after insert) {
+    if(Trigger.isBefore && Trigger.isInsert){
+        for(Account acc : Trigger.new){
+            if(String.isBlank(acc.Type)){
+                acc.Type = 'Prospect';
+            }    
+            if(String.isNotBlank(acc.ShippingStreet)){
+                acc.BillingStreet = acc.ShippingStreet;
+            }
+            if(String.isNotBlank(acc.ShippingCity)){
+                acc.BillingCity = acc.ShippingCity;
+            }
+            if(String.isNotBlank(acc.ShippingState)){
+                acc.BillingState = acc.ShippingState;
+            }
+            if(String.isNotBlank(acc.ShippingCountry)){
+                acc.BillingCountry = acc.ShippingCountry;
+            }
+            if(String.isNotBlank(acc.ShippingPostalCode)){
+                acc.BillingPostalCode = acc.ShippingPostalCode;    
+            }
```

Comment:
Address is really a complete unit, so I would check them each individually. Also you probably want to check the target billing fields are not null, not the source. It wont matter if the source field are blank if you copy them over, but you dont want to overwrite if billing address values are supplied.